editor.planx.uk/src/@planx/components/List/utils.tsx

```diff
@@ -74,14 +95,19 @@ export function sumIdenticalUnitsByDevelopmentType(
   return formattedSums;
 }
 
+interface FlattenOptions {
+  depth?: number;
+  path?: string | null;
+  separator?: string;
+};
+
 /**
  * Flattens nested object so we can output passport variables like `{listFn}.{itemIndexAsText}.{fieldFn}`
  *   Adapted from https://gist.github.com/penguinboy/762197
  */
 export function flatten<T extends Record<string, any>>(
-  object: T,
-  path: string | null = null,
-  separator = ".",
+    object: T,
+    { depth = Infinity, path = null, separator = ".", }: FlattenOptions = {}
```

Comment:
Without this change (adding `depth` argument), this will output keys in for format `${componentFn}.${number}.${fieldFn}.{$number}` via the existing recursive approach, For example - 

```json
{
  "residentialUnits.one.compliance.one": "m42",
  "residentialUnits.one.compliance.two": "m42a",
  "residentialUnits.one.compliance.three": "m42b",
}
```

I think the right approach isto output an array of responses, as we do with other checklists.This results in output formatted like this (see tests also) - 

```json
{
  "residentialUnits.one.compliance": ["m42", "m42a", "m42b"]
}
```